tmt/steps/provision/bootc.py

```diff
@@ -0,0 +1,364 @@
+import dataclasses
+import os
+from typing import TYPE_CHECKING, Optional, cast
+
+import tmt
+import tmt.base
+import tmt.hardware
+import tmt.log
+import tmt.steps
+import tmt.steps.provision
+import tmt.steps.provision.testcloud
+import tmt.utils
+from tmt.steps.provision.testcloud import GuestTestcloud
+from tmt.utils import field
+from tmt.utils.templates import render_template
+
+if TYPE_CHECKING:
+    from tmt.hardware import Size
+
+DEFAULT_TMP_PATH = "/var/tmp/tmt"  # noqa: S108
+
+DEFAULT_IMAGE_BUILDER = "quay.io/centos-bootc/bootc-image-builder:latest"
+CONTAINER_STORAGE_DIR = tmt.utils.Path("/var/lib/containers/storage")
+
+PODMAN_MACHINE_NAME = 'podman-machine-tmt'
+PODMAN_ENV = tmt.utils.Environment.from_dict(
+    {"CONTAINER_CONNECTION": f'{PODMAN_MACHINE_NAME}-root'})
+
+DEFAULT_PODMAN_MACHINE_CPU = 2
+DEFAULT_PODMAN_MACHINE_MEM: 'Size' = tmt.hardware.UNITS('2048 MB')
+DEFAULT_PODMAN_MACHINE_DISK_SIZE: 'Size' = tmt.hardware.UNITS('50 GB')
+
+CONTAINER_TEMPLATE = """
+FROM {{ base_image }}
+
+RUN <<EOF
+set -euo pipefail
+
+# install dependencies required by tmt
+dnf -y install cloud-init rsync
+ln -s ../cloud-init.target /usr/lib/systemd/system/default.target.wants
+dnf clean all
+
+# add the scripts in /var/lib/tmt/scripts to the PATH
+touch /etc/environment && \
+echo "export PATH=$PATH:/var/lib/tmt/scripts" >> /etc/environment && \
+
+# create a simple webserver used by tmt to verify the VM is running
```

Comment:
@frantisekz @ckyrouac I wonder if this is needed now that testcloud was fixed to be able to start this sever by itself:

https://github.com/teemtee/testcloud/pull/6/files

Reply:
nice! Removed it